deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Builds.java

```diff
@@ -210,6 +210,15 @@ public Response publish(
                 "Artifact URL points to unapproved location.");
         }
 
+        // Initialize the list of Source Control Managers
+        sourceControlManagerProxy.getSCMs();
+
+        // Check if build SCM is approved via allow list of SCMs
+        if (!sourceControlManagerProxy.approved(buildBean.getScm())) {
```

Comment:
maybe `approved` function name could be clearer from the perspective of the reader in this module, e.g. `hasScmType`

Reply:
Agreed. I will use` hasSCMType()`